internal/pkg/install/prepare.go

```diff
@@ -22,6 +22,35 @@ import (
 	"github.com/autonomy/talos/internal/pkg/userdata"
 	"github.com/autonomy/talos/internal/pkg/version"
 	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	// DefaultSizeRootDevice is the default size of the root partition.
+	// TODO(andrewrynhard): We should inspect the tarball's uncompressed size and dynamically set the root partition's size.
```

Comment:
re: TODO
This could get a bit interesting if we're going to potentially have a different size every time we have a new release.

Reply:
Yeah, its mostly a note as a possibility for now. But it feels like we should be able to dynamically set partition sizes. Something to dig into in the future.